### PR TITLE
Handle all Podman states

### PR DESCRIPTION
**Pull request by maturar**
Issues: #123 

# Definition of Done

The PR shall be merged only if all items mentioned in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been followed. In case an item is not applicable as described, please provide a short explanation in the description.

- [x] All steps in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been handled

### CODE REVIEW

**Review by lingnoi — APPROVED** [review]
LGTM 👍

**Comment by maturar** [comment]
I have also asked @krucod3 to review this PR, because this change may have effect to #48.

File: agent/doc/swdesign/README.md [hunk 1]
```diff
@@ -1492,15 +1492,22 @@ The `PodmanStateGetter` shall map the workload state returned by the Podman into
...
 | Stopping               |         -          |    Stopping    |
+| Stopped                |         -          |    Stopping    |
+| Removing               |         -          |    Stopping    |
 | (anything else)        |         -          |    Unknown     |
 
 Comment:
 The Podman also supports "pod states". This table shows the container states only.
+The container states `Stopped` and `Removing` are mapped to the workload state `Stopping`,
+because they are considered as transition states from the state `Succeeded` or `Running` into `Removed`.
+It is not desired to map them into `Unknown`. It would have meant that the Ankaios would have reported the state `Unknown` during workload deletion.
```
Comment (krucod3) [inline agent/doc/swdesign/README.md:1510]:
```suggestion
Mapping the state `Stopped` to a value different than `Stopping` would result in Ankaios reporting shortly some other state during workload deletion.
We are assuming that Ankaios workloads are not managed by an entity different than Ankaios and the container state `Stopped` is only transitionary.
```

**Review by krucod3 — COMMENTED** [review]
👍 Great! Now we also have the reasoning. I just added some clarifications in the comment of the requirement.

**Review by krucod3 — APPROVED** [review]
LGTM 👍

**Pull request merged**